### ISSUE TITLE
ACT Add additional fail reason code to not report failure to ACT test case when rule is more complex 

### ISSUE THREAD

**Issue by philljenkins** [IBMa/equal-access#1116]
Checker Rule: **WCAG21_Input_Autocomplete**
ACT Rule: **`autocomplete` attribute has valid value** ID: 73f2c2
[Passed Example 8](https://www.w3.org/WAI/content-assets/wcag-act-rules/testcases/73f2c2/35d0948a2cb309923e9a7cc5dd99a8ffc975088b.html) Expected passed, but Checker returned earl:failed 

Fail_1:  "The 'autocomplete' attribute's token(s) are not appropriate for the input form field"
is a correct message, but over reports what the ACT rule is expecting, which is only that the token(s) is programmatically determinable.  The Checker rule is checking both, if its determinable and if it is appropriate. ACT is only (initially) testing if it is determinable.

### Fix
Create an additional fail code so that test case 8 would pass when it is determinable, but fails when it isn't appropriate.  The fail case would not be mapped to this ACT rule ID: 73f2c2.

### Background and examples

This example rule shows how it maps to two different ACT rules (23a2a8 and 7d6734) and the second one includes different fail code mappings.
only "Fail_2" is mapped/reported against the ACT rule 
[HAAC_Aria_ImgAlt.ts](https://github.com/IBMa/equal-access/blob/master/accessibility-checker-engine/src/v4/rules/HAAC_Aria_ImgAlt.ts)
` act: ["23a2a8", {`
`        "7d6734": {`
`            "Pass_0": "pass",`
`            "Fail_1": "inapplicable",`
`            "Fail_2": "fail",`
`            "Fail_3": "inapplicable" `

#### All rules triggered by the ACT test case:
- "WCAG21_Input_Autocomplete:Fail_1:FAIL", [this issue]
- "WCAG20_Body_FirstASkips_Native_Host_Sematics:Fail_1:FAIL",
- "Rpt_Aria_OrphanedContent_Native_Host_Sematics:Fail_1:FAIL",
- "RPT_Html_SkipNav:Potential_1:POTENTIAL"

**Comment by philljenkins** [comment]
The code snippet captured by the Checker for the [Passed Example 8](https://www.w3.org/WAI/content-assets/wcag-act-rules/testcases/73f2c2/35d0948a2cb309923e9a7cc5dd99a8ffc975088b.html) is 
`"snippet": "<input autocomplete=\"bday-day\" type=\"tel\" name=\"bdayday\">",`
so the Checker rule is correctly detecting that "bday-day\" is not appropriate for a telephone type input,
but the ACT rule 73f2c2 is only requiring to detect that the HTML autocomplete attribute has a correct value.
>... element with an autocomplete [attribute value](https://wai-wcag-act-rules.netlify.app/standards-guidelines/act/rules/73f2c2/#attribute-value) that is neither empty ("") nor only [ASCII whitespace](https://infra.spec.whatwg.org/#ascii-whitespace), except if one or more of the following is true:
toggle: the autocomplete attribute consists of a single token that is an [ASCII case-insensitive](https://infra.spec.whatwg.org/#ascii-case-insensitive) match for the string off or the string on; or
disabled: the element is a [disabled element](https://wai-wcag-act-rules.netlify.app/standards-guidelines/act/rules/73f2c2/#disabled-element); or
fixed value: the element is an input element with a type [attribute value](https://wai-wcag-act-rules.netlify.app/standards-guidelines/act/rules/73f2c2/#attribute-value) of hidden, button, submit or reset; or
hidden: the element is not [visible](https://wai-wcag-act-rules.netlify.app/standards-guidelines/act/rules/73f2c2/#visible), and not [included in the accessibility tree](https://wai-wcag-act-rules.netlify.app/standards-guidelines/act/rules/73f2c2/#included-in-the-accessibility-tree); or
static: the element is not part of [sequential focus navigation](https://html.spec.whatwg.org/#sequential-focus-navigation) and has a [semantic role](https://wai-wcag-act-rules.netlify.app/standards-guidelines/act/rules/73f2c2/#semantic-role) that is not a [widget role](https://www.w3.org/TR/wai-aria-1.1/#widget_roles).

**Comment by aliunwala** [comment]
Triage:
We need to create a new fail code and map it to the ACT test cases. 

Similar issue to:
https://github.com/IBMa/equal-access/issues/1095

**Comment by shunguoy** [comment]
@philljenkins  the [Passed Example 8](https://www.w3.org/WAI/content-assets/wcag-act-rules/testcases/73f2c2/35d0948a2cb309923e9a7cc5dd99a8ffc975088b.html) even violates HTML 5 for the reason "The autofill field name bday-day is not allowed in this context", so the issue is on the ACT, rather than the checker.
